tests/loginPage.ts

```diff
@@ -0,0 +1,29 @@
+import { Credential } from "./creditintial";
+import { Locator, Page } from "@playwright/test";
+
+export default class loginPage {
+  private pageDomain: Page;
+
+  usernameTextBox: Locator;
+  passwordTextBox: Locator;
+  loginButton: Locator;
+  invalidCredential: Locator;
+  credentialList: Credential[] = [
+    new Credential("Admin1", "passw", "fail"),
+    new Credential("Admin2", "password123", "fail"),
+    new Credential("User1", "userpass", "fail"),
+    new Credential("Admin", "admin123", "success"),
+  ];
+
+  constructor(pageDomain: Page) {
+    this.pageDomain = pageDomain;
+    this.initializeLocators();
+  }
+
+  initializeLocators() {
```

Comment:
no need for this please add the locators directly inside the constructor

Reply:
may I understand why it's not necessary ?!